api/client/build.go

```diff
@@ -168,8 +181,12 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	// Dockerfile which uses trusted pulls.
 	context = replaceDockerfileTarWrapper(context, newDockerfile, relDockerfile)
 
+	// clear the progress buffer if -q is provided.
+	if buff, ok := progBuff.(*bytes.Buffer); ok {
+		buff.Reset()
```

Comment:
I think the output should be the exact same if things fail since we never know which bit of data will help them debug the failure.